client/beefy/src/communication/request_response/incoming_requests_handler.rs

```diff
@@ -149,9 +149,14 @@ where
 	fn handle_request(&self, request: IncomingRequest<B>) -> Result<(), Error> {
 		// TODO (issue #12293): validate `request` and change peer reputation for invalid requests.
 
+		let hash = match self.client.block_hash(request.payload.begin).map_err(Error::Client)? {
+			Some(hash) => hash,
+			None => return Ok(()),
```

Comment:
If the block hash can't be found here, handler should also send an `Ok(())` response to this request.

Aka, just chain this block together with next block so that `maybe_encoded_proof = Ok(())` will be sent when there is no hash availalable to the client.

Reply:
good point, thanks!
fixed in [91a6db8](https://github.com/paritytech/substrate/pull/12602/commits/91a6db874427f34024464bd5afdb1009e2014444)